doc/contribute/guidelines.rst

```diff
@@ -544,6 +544,8 @@ exceptions:
 
 * The line length is 100 columns or fewer. In the documentation, longer lines
   for URL references are an allowed exception.
+* Variables must be defined at the beginning of a function, in "reverse
```

Comment:
I suspect this point may be controversial. If it is I can drop it.

Reply:
I'm afraid we do not follow such convention in many places

Reply:
Also, these are two independent requirements, so even if it was kept it should be split into two. It should also reference scope and not function, since I believe we encourage variables to be defined in the smallest possible scope. Anyway, I also think that this should likely just be dropped. If there's a place for "recommendations" (as opposed to requirements) in this document, maybe the variable ordering (the reverse xmas tree thing) could be mentioned somewhere.

Reply:
I agree, let's drop this for now, I don't want to delay the current PR with these changes.

Reply:
If possible, would it be possible to set two levels, Required and Advisory, like MISRA-C?
I think that if reverse Christmas tree is made a rule, there will be many cases where it cannot be followed, but I agree with making it clear that it is a desirable norm.
(I know it would be a lot of work...)